pandas/core/series.py

```diff
@@ -235,6 +234,18 @@ def __init__(self, data=None, index=None, dtype=None, name=None,
                 if not is_list_like(data):
                     data = [data]
                 index = com._default_index(len(data))
+            elif is_list_like(data):
+
+                # a scalar numpy array is list-like but doesn't
+                # have a proper length
+                try:
+                    if len(data) != 1 and len(index) != len(data):
```

Comment:
still not convinced about this, what fails for len(data)

Reply:
If I remove `len(data) != 1` and run `python -m pytest pandas/tests/io/formats/test_style.py` I get:
```
                try:
                    if len(index) != len(data):
                        raise ValueError(
                            'Length of passed values is {val}, '
                            'index implies {ind}'
>                           .format(val=len(data), ind=len(index)))
E                           ValueError: ('Length of passed values is 1, index implies 2', 'occurred at index A')

pandas/core/series.py:246: ValueError
```
I should add a test for this case in test_constructors.

Reply:
ok add a test in test_constructors. which is this failing on in test_style?

Reply:
test_apply_subset.  Input to the Series constructor is:
```
data = ['color: baz'],
index = RangeIndex(start=0, stop=2, step=1), 
dtype = None, 
name = 'A', 
copy = False, 
fastpath = False
```

Reply:
So this should raise! if the data is a scalar this is ok, but we can't broadcast a list like that (well we can, but we shoudn't)
```
In [2]: data = ['color: baz']
   ...: index = pd.RangeIndex(start=0, stop=2, step=1)
   ...: 

In [3]: pd.Series(data, index)
Out[3]: 
0    color: baz
1    color: baz
dtype: object
```